cryptography/src/lib.rs

```diff
@@ -51,14 +51,13 @@ pub trait Scheme: Clone + Send + Sync + 'static {
 
     /// Sign the given message.
     ///
-    /// The message should not be hashed prior to calling this function. If a particular
-    /// scheme requires a payload to be hashed before it is signed, it will be done internally.
+    /// The message should not be hashed prior to calling this function. If a particular scheme
+    /// requires a payload to be hashed before it is signed, it will be done internally.
     ///
-    /// To protect against replay attacks, it is required to provide a namespace
-    /// to prefix any message. This ensures that a signature meant for one context cannot be used
-    /// unexpectedly in another (i.e. signing a message on the network layer can't accidentally
-    /// spend funds on the execution layer).
-    fn sign(&mut self, namespace: &[u8], message: &[u8]) -> Signature;
+    /// A namespace should be used to prevent replay attacks. It will be prepended to the message so
```

Comment:
nit: Wonder if we should talk exactly about what the prepended part looks like (or at least point to `union_unique`)?